catroid/src/test/java/org/catrobat/catroid/test/content/actions/DeleteItemOfUserListActionTest.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions
+
+import org.catrobat.catroid.content.ActionFactory
+import org.catrobat.catroid.content.actions.DeleteItemOfUserListAction
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.UserList
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class DeleteItemOfUserListActionTest(
+    private val testName: String,
+    private val listIndexToDelete: List<Formula>,
+    private var initList: List<Any>,
+    private var expectedList: List<Any>
+) {
+
+    private lateinit var userList: UserList
+    private lateinit var actionFactory: ActionFactory
+    private lateinit var deleteaction: DeleteItemOfUserListAction
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun parameters() = listOf(
+            arrayOf("DeleteFirstEntry", listOf(Formula(1)), mutableListOf(1, 2, 3), listOf(2, 3)),
+            arrayOf("DeleteMiddleEntry", listOf(Formula(2)), mutableListOf(1, 2, 3), listOf(1, 3)),
+            arrayOf("DeleteMultipleEntries", listOf(Formula(1), Formula(2)), mutableListOf(1, 2, 3), listOf(2)),
+            arrayOf("DeleteAllEntries", listOf(Formula(1), Formula(1)), mutableListOf("First", 2), emptyList<Any>()),
+            arrayOf("DeleteInvalidEntries", listOf(Formula(3), Formula(0), Formula(-1)), mutableListOf(1, 2), listOf(1, 2)),
+            arrayOf("DeleteNotANumberEntry", listOf(Formula(Double.NaN)), mutableListOf(1, 2), listOf(2)),
```

Comment:
```suggestion
            arrayOf("DeleteNotANumberEntry", listOf(Formula(Double.NaN)), mutableListOf(1, 2), listOf(1, 2)),
```
The removal is actually a bug, an incorrect behavior of the implementation. The correct behavior should be that the item is not removed (note: it is correctly implemented in Scratch and in Catty). Please correct the bug (preferred) or remove this line, and also in the CLT, here:

![image](https://user-images.githubusercontent.com/1747888/108604566-02e58780-73af-11eb-8be6-706609d0311c.png)

Reply:
The Bug is now fixed and the CLT is accordingly modified.